File: front-end/src/app/modelo/asistencia.ts
import { EstadoAsistencia } from './estadoasistencia';
import { Subprocesos } from './subprocesos';
import { Usuario } from './usuario';

export class Asistencia {
  asisId?: number;
  asisNombre?: string;
  asisDpto?: string;
  asisNoLector?: number;
  asisFechaHora?: Date | string | null;
  asisEstado?: string;
  asisLocacionId?: number;
  asisIdNumero?: number;
  asisCodTrabajo?: number;
  asisVerificaCod?: string;
  asisNoTarjeta?: number;
  asisNombreArchivo: string;
  asisFechaArchivo: Date;
  usuId: Usuario;
  asisEstadoStr: string;


  constructor(
    asisId?: number,
    asisNombre?: string,
    asisDpto?: string,
    asisNoLector?: number,
    asisFechaHora?: Date | string | null,
    asisEstado?: string,
    asisLocacionId?: number,
    asisIdNumero?: number,
    asisCodTrabajo?: number,
    asisVerificaCod?: string,
    asisNoTarjeta?: number,
    asisNombreArchivo?: '',
    asisFechaArchivo?: Date,
    usuId?: Usuario,
    asisEstadoStr?: string,
  ) {
    (this.asisId = asisId || 0),
      (this.asisNombre = asisNombre || ''),
      (this.asisDpto = asisDpto || '');
    (this.asisNoLector = asisNoLector || 0),
      (this.asisFechaHora = asisFechaHora || new Date()),
      (this.asisEstado = asisEstado = ''),
      (this.asisLocacionId = asisLocacionId || 0),
      (this.asisIdNumero = asisIdNumero || 0),
      (this.asisCodTrabajo = asisCodTrabajo || 0),
      (this.asisVerificaCod = asisVerificaCod || '');
    (this.asisNoTarjeta = asisNoTarjeta || 0);
    (this.asisNombreArchivo = asisNombreArchivo || '');
    (this.asisFechaArchivo = asisFechaArchivo || new Date());
    (this.usuId = usuId || new Usuario);
    (this.asisEstadoStr = asisEstadoStr || '')
  }
}

File: back-end/app/src/main/java/com/senescyt/app/model/Asistencia.java
package com.senescyt.app.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.sql.Timestamp;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "asistencia")
public class Asistencia implements Serializable {

    /**
     *
     */
    private static final long serialVersionUID = 1L;
    /**
     *
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "asisId")
    private Long asisId;

    private String asisNombre;

    private String asisDpto;

    private Long asisNoLector;

    private String asisFechaHora;

    private String asisEstado;

    private Long asisLocacionId;

    private Long asisIdNumero;

    private Long asisCodTrabajo;

    private String asisVerificaCod;

    private Long asisNoTarjeta;

    private String asisEstadoStr;// Puntual o Retraso

    private String asisNombreArchivo;

    private Timestamp asisFechaArchivo;



    @ManyToOne
    @JoinColumn(name = "usuId", referencedColumnName = "usuId")
    private Usuario usuId;

//    @ManyToOne
//    @JoinColumn(name = "estId", referencedColumnName = "estId")
//    private EstadoAsistencia estadoAsistencia;

}
